packages/perf/lib/index.d.ts

```diff
@@ -384,7 +384,22 @@ export namespace FirebasePerformanceTypes {
      * ```
      */
     isPerformanceCollectionEnabled: boolean;
+    /**
+     * Determines whether to collect 'out of the box' (i.e already setup for Firebase Performance) events.
+     */
+    isInstrumentationEnabled: boolean;
+    /**
+     * Determines whether performance monitoring is enabled or disabled.
+     *
+     * #### Example
+     *
+     * ```js
+     * const isEnabled = firebase.perf().isDataCollectionEnabled;
+     * console.log('Performance collection enabled: ', isEnabled);
+     * ```
+     */
 
+    isDataCollectionEnabled: boolean;
```

Comment:
These deviate from the firebase-js-sdk API where they are properties that are settable, but are not getters/setters - that (javascript properties) is something we do not do very much in react-native-firebase but we do it in a couple places, and this is one where we have to in order to be firebase-js-sdk compatible

https://github.com/firebase/firebase-js-sdk/blob/49ee786f2b022e65aef45693e1a8b546d889ec10/packages/performance-types/index.d.ts#L30-L34

Reply:
@mikehardy  How will that work with existing API?

Currently:
```js

  get isPerformanceCollectionEnabled() {
    return this._isPerformanceCollectionEnabled;
  }

  get isInstrumentationEnabled() {
    return this._isInstrumentationEnabled;
  }

  isDataCollectionEnabled() {
    return this.isPerformanceCollectionEnabled;
  }

  setDataCollectionEnabled(enabled) {
    return this.setPerformanceCollectionEnabled(enabled);
  }

  setPerformanceCollectionEnabled(enabled) {
    if (!isBoolean(enabled)) {
      throw new Error(
        "firebase.perf().setPerformanceCollectionEnabled(*) 'enabled' must be a boolean.",
      );
    }
```

Just so I understand. Wouldn't I have to remove existing methods `setPerformanceCollectionEnabled` & `isPerformanceCollectionEnabled` if we're moving to a simple `isDataCollectionEnabled` property without getter/setter? This would mean we wouldn't have the type checking for `enabled` here:

```js
  setPerformanceCollectionEnabled(enabled) {
    if (!isBoolean(enabled)) {
      throw new Error(
        "firebase.perf().setPerformanceCollectionEnabled(*) 'enabled' must be a boolean.",
      );
    }
```